consensus/bor/heimdall/client_test.go

```diff
@@ -1,9 +1,317 @@
 package heimdall
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+	"net/http"
+	"sync"
+	"sync/atomic"
 	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/bor/heimdall/checkpoint"
+
+	"github.com/stretchr/testify/require"
 )
 
+var maxPortCheck int32 = 100
+
+// HttpHandlerFake defines the handler functions required to serve
+// requests to the mock heimdal server for specific functions. Add more handlers
+// according to requirements.
```

Comment:
Can we try https://pkg.go.dev/net/http/httptest ?

Reply:
Will check it.

Reply:
Resolving this as per our conversation. This package is used to create mock requests and test the server side logic. As we have a heimdall client which has additional methods, it's good to test things using those methods. In order to do that, the easiest way for now is a mock server.